subprojects/docs/docs.gradle

```diff
@@ -447,17 +280,109 @@ tasks.register("checkstyleApi", Checkstyle) {
+    }
+    outputDir = userguideIntermediateOutputDir
+    // Verbose mode reports broken links. See asciidoctor/asciidoctor-gradle-plugin#233
+    requires file('src/docs/ruby/verbose-mode.rb')
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends = ['html5']
+
+    attributes stylesdir      : '../css/',
+        stylesheet            : 'manual.css',
+        imagesdir             : 'images',
+        nofooter              : true,
+        icons                 : null,
+        'note-caption'        : '&#128161;',
+        'tip-caption'         : '&#128161;',
+        'important-caption'   : '&#10071;️',
+        'caution-caption'     : '&#9888;',
+        'warning-caption'     : '&#9888;',
+        'source-highlighter'  : 'prettify',
+        sectanchors           : true,
+        sectlinks             : true,
+        linkattrs             : true,
+        reproducible          : '',
+        lang                  : 'en-US',
+        encoding              : 'utf-8',
+        idprefix              : '',
+        toc                   : 'auto',
+        toclevels             : 1,
+        'toc-title'           : 'Contents',
+        'gradle-home-url'     : 'https://gradle.org',
+        'java-api-url'        : 'https://docs.oracle.com/javase/7/docs/api',
+        'ant-manual-url'      : 'https://ant.apache.org/manual',
+        'guides-url'          : 'https://guides.gradle.org',
+        'gradle-version'      : version,
+        'groovy-dsl-path'     : "../dsl",
+        'javadoc-path'        : "../javadoc",
+        'samples-path'        : "../../samples"
+}
+
+def userguidePdf = tasks.register("userguidePdf", AsciidoctorTask) {
+    dependsOn defaultImports
+    sourceDir = userguideSrcDir
+    sources { include 'userguide_single.adoc' }
+    outputDir = userguidePdfOutputDir
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends 'pdf'
+
+    attributes \
+        lang                : 'en-US',
+        encoding            : 'utf-8',
+        toc                 : 'preamble',
+        toclevels           : 3,
+        imagesdir           : 'images',
+        'source-highlighter': 'coderay',
+        'gradle-home-url'   : 'https://gradle.org',
+        'java-api-url'      : 'https://docs.oracle.com/javase/7/docs/api',
+        'ant-manual-url'    : 'https://ant.apache.org/manual',
+        'guides-url'        : 'https://guides.gradle.org',
+        'gradle-version'    : version,
+        'groovy-dsl-path'   : "https://gradle.org/$version/dsl",
+        'javadoc-path'      : "https://gradle.org/$version/javadoc",
+        'samples-path'      : "../../samples"
+}
+
+// Avoid overlapping outputs by copying exactly what we want from other intermediate tasks
+def userguide = tasks.register("userguide", Copy) {
+    dependsOn userguideAsciidoc, userguidePdf
+    description = 'Generates the userguide HTML and PDF'
     group = 'documentation'
+
+    outputs.dir userguideOutputDir
```

Comment:
`output.dir` is unnecessary here since we already have `into userguideOutputDir`.